Runtime/Scripts/Api/V1/OpenAiApiV1.cs

```diff
@@ -54,6 +59,7 @@ public OpenAiApiV1(SAuthArgsV1 authArgs)
             Engines = new EnginesResourceV1(this);
             Files = new FilesResourceV1(this);
             Classifications = new ClassificationsResourceV1(this);
+            ChatCompletions = new ChatCompletionsResourceV1(this);
```

Comment:
Since the `ChatCompletions` resource is accessed via the `EnginesResource` (Specifically in the `EngineResource`), it should not appear in this class. It should appear as a child of the `EngineResource`

I just took a look at the API quickly, and I see some major concepts have changed. Engines is depreceated. 

However, I think that in this PR, the code should remain as consistent as possible with the existing Resources, so that when I or someone else does a refactor based on the new API concepts, everything can be moved in the same way. 

So, to finalize this comment.
* Remove the `ChatCompletions` property from this class. Accessing this resource should be done through `OpenAiApiV1.Engines.Engine("engine_id").Chat.Completions`

Reply:
I originally tried to treat chat completions as an engine consistent with prompt completions but chat completions cannot be accessed via the endpoint engines/<engine_id>/chat/completions. I think chat must be accessed via an endpoint that does not include the engine; the model is passed as part of the JSON payload, example below. So that's the rationale for adding the chat resource to the root of the API rather than as an engine. Let me know your thoughts when you have a chance. I'll look into your other PR change requests.

curl https://api.openai.com/v1/chat/completions \
  -H "Content-Type: application/json" \
  -H "Authorization: Bearer $OPENAI_API_KEY" \
  -d '{
    "model": "gpt-3.5-turbo",
    "messages": [{"role": "user", "content": "Hello!"}]
  }'

Reply:
Ok, if that is the case (which I can see from reading the API specs), then we still need two objects instead of one. Since the endpoint is `chat/completions` to stay consistent we want the access to work as `OpenAiApiV1.Chat.Completions` rather than `OpenAiApiV1.ChatCompletions`. That way if there are more `chat/` endpoints in the future they will be easy to add. 

I know there is already a `CompletionsResourceV1` object so naming the new one `ChatCompletionsResourceV1` will work. But in the `ChatResourceV1` the property name should still be `Completions`. I.E.:

```csharp
public class ChatResourceV1 : AApiResource<OpenAiApiV1>
{
   ...
   public ChatCompletionsResourceV1 Completions { get; private set;} // not ChatCompletionsResourceV1 ChatCompletions
   ...
}
```

Reply:
Got it, will make this change. Thank you for taking the time to work with me on this James.

Reply:
Refactored OpenAiApiV1.ChatCompletions into OpenAiApiV1.Chat.Completions. Getting there...